cni/pkg/ambient/informers.go

```diff
@@ -103,20 +116,20 @@ func (s *Server) Reconcile(input any) error {
 		wasEnabled := oldPod.Annotations[constants.AmbientRedirection] == constants.AmbientRedirectionEnabled
 		nowEnabled := ambientpod.PodZtunnelEnabled(ns, newPod)
 		if wasEnabled && !nowEnabled {
-			log.Debugf("Pod %s no longer matches, removing from mesh", newPod.Name)
+			log.Debugf("Pod no longer matches, removing from mesh")
 			s.DelPodFromMesh(newPod)
 		}
 
 		if !wasEnabled && nowEnabled {
-			log.Debugf("Pod %s now matches, adding to mesh", newPod.Name)
+			log.Debugf("Pod now matches, adding to mesh")
 			s.AddPodToMesh(pod)
 		}
```

Comment:
```suggestion
		}
		
+		if nowEnabled && !IsPodInIpset(pod) {
+			// This can happen if a node cleanup happens as part of a ztunnel recycle,
+			// cleaning up the node-level ipset with all the pod IPs
+			// If this happens we re-queue everything for reconciliation, and need to
+			// make sure existing pods get re-added to the ipset if they aren't already there.
+			log.Debugf("Pod %s is enabled but not in ipset, (re)adding to mesh", newPod.Name)
+			s.AddPodToMesh(pod)
 		}
```